docs/spec/spec.md

```diff
@@ -0,0 +1,416 @@
+# Specification: Ballerina SAP JCo Connector
+
+_Authors_: @RDPerera\
+_Reviewers_: @daneshk @aashikam @shafreenAnfar\
+_Created_: 2024/08/28\
+_Updated_: 2024/09/06\
+_Edition_: Swan Lake
+
+## Introduction
+
+This is the specification for the SAP JCo connector library of [Ballerina language](https://ballerina.io/), which is used for integrating Ballerina applications with SAP systems.
+
+The SAP JCo connector specification has evolved and may continue to evolve in the future. The released versions of the specification can be found under the relevant GitHub tag.
+
+If you have any feedback or suggestions about the connector, start a discussion via a GitHub issue or in the Discord server. Based on the outcome of the discussion, the specification and implementation can be updated. Community feedback is always welcome. Any accepted proposal, which affects the specification is stored under /docs/proposals. Proposals under discussion can be found with the label type/proposal in GitHub.
+
+The conforming implementation of the specification is released and included in the distribution. Any deviation from the specification is considered a bug.
+
+## Contents
+
+1. [Overview](#1-overview)
+2. [Components](#2-components)
+   - 2.1 [RFC Client](#21-rfc-client)
+     - 2.1.1 [Configurations](#211-configurations)
+       - 2.1.1.1 [DestinationConfig](#2111-destinationconfig)
+       - 2.1.1.2 [AdvancedConfig](#2112-advancedconfig)
+     - 2.1.2 [Available Operations](#212-available-operations)
+       - 2.1.2.1 [Execute a Function Module via RFC](#2121-execute-a-function-module-via-rfc)
+       - 2.1.2.2 [Send IDocs to an SAP System](#2122-send-idocs-to-an-sap-system)
+     - 2.1.3 [Data Types](#213-data-type-mapping-between-ballerina-and-sap-jco)
+   - 2.2 [Listener](#22-listener)
+     - 2.2.1 [Configurations](#221-configurations)
+       - 2.2.1.1 [ServerConfig](#2211-serverconfig)
+       - 2.2.1.2 [AdvancedConfig](#2212-advancedconfig)
+     - 2.2.2 [Available Operations](#222-available-operations)
+       - 2.2.2.1 [Receive IDocs](#2221-receive-idocs)
+       - 2.2.2.2 [Error Handling in Listener](#2222-error-handling)
```

Comment:
IINM  we recently agreed to use sentence case for titles as a convention across all new library content. So I believe this should be applied to the spec as well. @ThisaruGuruge @daneshk wdyt?

Reply:
Yes, let's use sentence case

Reply:
Changed with [0553b90](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/56/commits/0553b901a67bf7a546060ea221052c9ae955cb00)